src/config/session.c

```diff
@@ -80,6 +83,79 @@ read_environment_file(const char *filename)
 	return true;
 }
 
+static char *
+valid_env_path(const char *prefix, const char *name)
+{
+	if (!prefix || string_null_or_empty(name)) {
+		return NULL;
+	}
+
+	/* Valid environment files always end in '.env' */
+	const char *suffix = strrchr(name, '.');
+	if (!suffix || strcmp(suffix, ".env") != 0) {
+		return NULL;
+	}
+
+	char *full_path = strdup_printf("%s/%s", prefix, name);
+	if (!full_path) {
+		return NULL;
+	}
+
+	/* Valid environment file must be a regular file */
+	struct stat statbuf;
+	if (stat(full_path, &statbuf) == 0 && S_ISREG(statbuf.st_mode)) {
+		return full_path;
+	}
```

Comment:
Is a symlink counting as regular file as well? If not we may also want to allow them here.

Reply:
In my testing, a broken symlink caused no problems, and a valid symlink was followed as expected.